firmware/application/apps/ui_fileman.cpp

```diff
@@ -403,11 +403,17 @@ void FileManagerView::refresh_widgets(const bool v) {
     set_dirty();
 }
 
-void FileManagerView::on_rename() {
+void FileManagerView::on_rename(std::string_view hint) {
     auto& entry = get_selected_entry();
-    name_buffer = entry.path.filename().string();
-    uint32_t cursor_pos = (uint32_t)name_buffer.length();
 
+    // Append the hint to the filename stem as a rename suggestion.
+    name_buffer = entry.path.stem().string();
+    if (!hint.empty())
+        name_buffer += "_" + std::string{hint};
+    name_buffer += entry.path.extension().string();
```

Comment:
Is there a maximum file name length that we need to worry about?

Reply:
Realistically, no. The UI has some arbitrary limit on the length in the text_prompt, but we should really just get rid of all that.
The FS also has like MAX_PATH limits but that's like 256 or something we're not likely to hit.